### PR TITLE
Delayed metrics reroll config

### PR DESCRIPTION
**Pull request by ChandraAddala**
Introducing a new configuration, DELAYED_METRICS_REROLL_GRANULARITY, using which we control the granularity upto which we re-roll only the delayed metrics instead of entire slot.

### CODE REVIEW

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/rollup/SlotKey.java [hunk 1]
```diff
@@ -135,6 +136,33 @@ public static SlotKey parse(String string) {
     }
 
     /**
+     * Same as the method {@link #getChildrenKeys()} except this method returns only
+     * the children corresponding to the destination granularity.
+     *
+     * For example, a slot key of granularity
+     * {@link Granularity#MIN_1440 MIN_1440}, for destination granularity of
+     * {@link Granularity#MIN_60 MIN_60}, will have 6*4=24 children.
```
Comment (shintasmith) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/rollup/SlotKey.java]: Curious, Is this a normal example? I mean, do we normally ask from 1 day (1440m) granularity to 1 hr (60m) granularity, skipping the 240m granularity? 

Also, the term "children" is not as clear, but I realized it's probably been used in other places. At first I thought 1440m is a child of the 240m. I don't know if it's possible to change to "Finer" and "Coarser"

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/rollup/SlotKey.java [hunk 2]
```diff
@@ -135,6 +136,33 @@ public static SlotKey parse(String string) {
...
+     *
+     * @param destGranularity
+     * @return
+     */
+    public Collection<SlotKey> getChildrenKeys(Granularity destGranularity) {
+
+        if (!getGranularity().isCoarser(destGranularity)) {
+            throw new IllegalArgumentException("Current granularity must be coarser than the destination granularity");
```
Comment (shintasmith) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/rollup/SlotKey.java]: Can you print the 2 granularities involved in the Exception msg?

File: CHANGES.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 ## IN PROGRESS
 * Upgraded datastax driver to 3.1.2
+* Added new configuration option DELAYED_METRICS_REROLL_GRANULARITY
```
Comment (ChandraAddala) [inline CHANGES.md]: Once everything is merged to master, I plan on writing a wiki page just for delayed metrics explaining various configs and how to tweak them.

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/rollup/SlotKey.java (hunk 1 above)
Comment (ChandraAddala) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/rollup/SlotKey.java]: There is getChildrenKeys above which would list children of all sublevels. This method uses that and only returns the level we are interested in. I just followed the terminology used there.

**Pull request merged**